core/src/main/scala/org/bitcoins/core/psbt/PSBTHelper.scala

```diff
@@ -0,0 +1,225 @@
+package org.bitcoins.core.psbt
+
+import org.bitcoins.core.crypto.{ECDigitalSignature, ECPublicKey, ExtPublicKey}
+import org.bitcoins.core.hd.BIP32Path
+import org.bitcoins.core.number.UInt32
+import org.bitcoins.core.protocol.CompactSizeUInt
+import org.bitcoins.core.protocol.script.{
+  ScriptPubKey,
+  ScriptSignature,
+  WitnessScriptPubKey
+}
+import org.bitcoins.core.protocol.transaction.{Transaction, WitnessTransaction}
+import org.bitcoins.core.psbt.GlobalPSBTRecord._
+import org.bitcoins.core.psbt.InputPSBTRecord._
+import org.bitcoins.core.psbt.PSBTGlobalKey._
+import org.bitcoins.core.psbt.PSBTInputKey._
+import org.bitcoins.core.script.crypto.HashType
+import org.bitcoins.core.serializers.script.RawScriptWitnessParser
+import org.bitcoins.core.util.BitcoinScriptUtil
+import scodec.bits.ByteVector
+
+import scala.annotation.tailrec
+
+object PSBTHelper {
+
+  private def parseKeyAndValue(
+      bytes: ByteVector): (Long, ByteVector, ByteVector) = {
```

Comment:
it would be nice to have a case class that wraps the `(Long, ByteVector, ByteVector)`, in general i don't really like returning tuples.